flink-runtime/src/main/java/org/apache/flink/runtime/operators/shipping/OutputEmitter.java

```diff
@@ -67,27 +67,11 @@ public OutputEmitter(ShipStrategyType strategy) {
 	 * @param comparator The comparator used to hash / compare the records.
 	 */
 	public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator) {
-		this(strategy, comparator, null, null);
+		this(strategy, comparator, null);
 	}
 	
-	/**
-	 * Creates a new channel selector that uses the given strategy (broadcasting, partitioning, ...)
-	 * and uses the supplied comparator to hash / compare records for partitioning them deterministically.
-	 * 
-	 * @param strategy The distribution strategy to be used.
-	 * @param comparator The comparator used to hash / compare the records.
-	 * @param distr The distribution pattern used in the case of a range partitioning.
-	 */
-	public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator, DataDistribution distr) {
-		this(strategy, comparator, null, distr);
-	}
-	
-	public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator, Partitioner<?> partitioner) {
-		this(strategy, comparator, partitioner, null);
-	}
-		
 	@SuppressWarnings("unchecked")
-	public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator, Partitioner<?> partitioner, DataDistribution distr) {
+	public OutputEmitter(ShipStrategyType strategy, TypeComparator<T> comparator, Partitioner<?> partitioner) {
```

Comment:
I would keep the `DataDistribution` in the `OutputEmitter`. 
We can later adapt this such that we can set the `DataDistribution` at execution time from the broadcasted set and avoid the partition ID assignment.

For now, we can simply set an Integer distribution with boundaries 1,2,3...,n.